app/src/main/java/org/mozilla/fenix/ServiceWorkerSupportFeature.kt

```diff
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.EngineSession.LoadUrlFlags
+import mozilla.components.concept.engine.serviceworker.ServiceWorkerDelegate
+import org.mozilla.fenix.ext.components
+
+/**
+ * Fenix own version of the `ServiceWorkerSupportFeature` from Android-Components
+ * which adds the ability to navigate to the browser before opening a new tab.
+ *
+ * Will automatically register callbacks for service workers requests and cleanup when [homeActivity] is destroyed.
+ *
+ * @param homeActivity [HomeActivity] used for navigating to browser or accessing various app components.
+ */
+class ServiceWorkerSupportFeature(
+    private val homeActivity: HomeActivity
+) : ServiceWorkerDelegate, DefaultLifecycleObserver {
+    override fun onDestroy(owner: LifecycleOwner) {
```

Comment:
I left comments on the A-C side. This is a bit of a new pattern where we have a global delegate tied to the application lifecycle but also need a reference to our activity. I think this solution is fine, but some changes over at A-C needed.

Reply:
Updated the AC patch. Thank you!